File: backend/entity/workshop/gameUser.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type GameUser struct {
	gorm.Model
	Username  string
	Password  string
	Email     string
	FirstName string
	LastName  string
	Birthday  time.Time

	OwnerGame []OwnerGame `gorm:"foreignKey:GameUserID"`
	Mod []Mod `gorm:"foreignKey:GameUserID"`
	ModRating []ModRating `gorm:"foreignKey:GameUserID"`
	
}

File: frontend/src/interfaces/GameUser.ts
import type { OwnerGame } from "./OwnerGame";
import type { Mod } from "./Mod";
import type { ModRating } from "./ModRating";

export interface GameUser {
  ID: number;
  Username: string;
  Password?: string;
  Email: string;
  FirstName?: string;
  LastName?: string;
  Birthday?: string;
  OwnerGame?: OwnerGame[];
  Mod?: Mod[];
  ModRating?: ModRating[];
}